onnxruntime/core/framework/kernel_registry_manager.cc

```diff
@@ -98,27 +95,24 @@ Status KernelRegistryManager::SearchKernelRegistry(const onnxruntime::Node& node
   Status status;
   {
     for (auto& registry : custom_kernel_registries_) {
-      *kernel_create_info = registry->TryFindKernel(node, "");  // the last argument is ignored
-      if (*kernel_create_info != nullptr) {
-        return Status::OK();
-      }
+      status = registry->TryFindKernel(node, std::string(), kernel_create_info);
+      if (status.IsOK()) return status;
     }
   }
 
   KernelRegistry* p = nullptr;
   auto iter = provider_type_to_registry_.find(ptype);
   if (iter != provider_type_to_registry_.end()) p = iter->second.get();
   if (p != nullptr) {
-    *kernel_create_info = p->TryFindKernel(node, "");  // the last argument is ignored
-    if (*kernel_create_info != nullptr) {
-      return Status::OK();
-    }
+    status = p->TryFindKernel(node, "", kernel_create_info);
```

Comment:
nit: we can possibly create a static empty string that we can pass to TryFindKernel in both cases instead of constructing every single time. This happens during initialization time, so may not be super important.

Reply:
Changed it to std::string().

Reply:
If we declare it as a static variable, we can't mark it as constexpr because it requires a non-trivial constructor and may cause initialization order problem in some rare cases.